burn-tensor/src/tensor/api/check.rs

```diff
@@ -374,6 +374,98 @@ impl TensorCheck {
         check
     }
 
+    pub fn gather<const D: usize>(dim: usize, shape: &Shape<D>, shape_indexes: &Shape<D>) -> Self {
```

Comment:
I think we should start putting some minimal doc strings for all public APIs. Good pattern to establish.

Reply:
I don't think this should be a public API, just pub(crate) I'll do the fix